File: hedgehog.cs
using System;

class Program
{
  static void Main()
  {
    int x = 0, y = 0;
    int dx = 0, dy = 1;
    string c = Console.ReadLine();

    while (!string.IsNullOrEmpty(c))
    {
      if (c == "R")
      {
        int temp = dx;
        dx = dy;
        dy = -temp;
      }
      else if (c == "L")
      {
        int temp = dx;
        dx = -dy;
        dy = temp;
      }
      else
      {
        x += dx;
        y += dy;
      }
      c = Console.ReadLine();
    }

    Console.WriteLine($"{x} {y}");
  }
}

File: hedgehog.go
package main

import "fmt"

func main() {
	x, y := 0, 0
	dx, dy := 0, 1
	var c string

	fmt.Scan(&c)

	for c != "" {
		if c == "R" {
			temp := dx
			dx = dy
			dy = -temp
		} else if c == "L" {
			temp := dx
			dx = -dy
			dy = temp
		} else {
			x += dx
			y += dy
		}
		fmt.Scan(&c)
	}

	fmt.Printf("%d %d\n", x, y)
}
